html_ui/Pages/VCockpit/Instruments/Airliners/B747_8/FMC/B747_8_FMC_MainDisplay.js

```diff
@@ -109,6 +200,93 @@ class B747_8_FMC_MainDisplay extends Boeing_FMC {
+        if (_event.indexOf("1_BTN_") !== -1 || _event.indexOf("2_BTN_") !== -1 || _event.indexOf("BTN_") !== -1 ||  _event.indexOf("3_") !== -1) {
+            const input = _event.replace("1_BTN_", "").replace("2_BTN_", "").replace("BTN_", "").replace("3_", "");
+            if (this.onInputAircraftSpecific(input)) {
+                return;
+            }
+            if (input === "INIT") {
+                this.onInit();
+            }
+            else if (input === "DEPARR") {
+                this.onDepArr();
+            }
+            else if (input === "ATC") {
+                this.onAtc();
+            }
+            else if (input === "FIX") {
+                this.onFix();
+            }
+            else if (input === "HOLD") {
+                this.onHold();
+            }
+            else if (input === "FMCCOMM") {
+                this.onFmcComm();
+            }
+            else if (input === "PROG") {
+                this.onProg();
+            }
+            else if (input === "MENU") {
+                this.onMenu();
+            }
+            else if (input === "NAVRAD") {
+                this.onRad();
+            }
+            else if (input === "PREVPAGE") {
+                this.onPrevPage();
+            }
+            else if (input === "NEXTPAGE") {
+                this.onNextPage();
+            }
+            else if (input === "SP") {
+                this.onSp();
+            }
+            else if (input === "DEL") {
+                this.onDel();
+            }
+            else if (input === "CLR") {
+                this.onClr();
+            }
+            else if (input === "CLR_Long") {
+                this.onClrLong();
+            }
+            else if (input === "DIV") {
+                this.onDiv();
+            }
+            else if (input === "DOT") {
+                this.inOut += ".";
+            }
+            else if (input === "PLUSMINUS") {
+                this.inOut += "-";
+            }
+            else if (input === "Localizer") {
+                this._apLocalizerOn = !this._apLocalizerOn;
+            }
+            else if (input.length === 2 && input[0] === "L") {
+                let v = parseInt(input[1]);
+                if (isFinite(v)) {
+                    if (this.onLeftInput[v - 1]) {
+                        this.onLeftInput[v - 1]();
+                    }
+                }
+            }
+            else if (input.length === 2 && input[0] === "R") {
+                let v = parseInt(input[1]);
+                if (isFinite(v)) {
+                    if (this.onRightInput[v - 1]) {
+                        this.onRightInput[v - 1]();
+                    }
+                }
+            }
+            else if (input.length === 1 && FMCMainDisplay._AvailableKeys.indexOf(input) !== -1) {
+                this.onLetterInput(input);
+            }
+            else {
+                console.log("'" + input + "'");
+            }
+        }
+    }
```

Comment:
What exactly does this add?

Reply:
It's a function from the generic maindisplay file.
I just added a replace to make the others CDUs work. 
Asobo changed the xml code of the CDUs in the last update.

Reply:
What do you mean other CDUs? Didn't it work before as well

Reply:
The FO and Pedestal CDU buttons stopped working after 1.12.13.

Reply:
Ah, I see. Maybe use a switch statement instead of the long if else chain, though.